spec/maths/statistics/vector_spec.rb

```diff
@@ -6,12 +6,46 @@
         @dv_with_nils = Daru::Vector.new [323, 11, 555, nil, 666, 234, 21, 666, 343, nil, 1, 2]
       end
 
+      let(:idx) { Daru::Index.new [:jon, :tyrion, :dany, :cersei, :jamie] }
+      let(:dv2) { Daru::Vector.new ["tyrion lannister", "arya stark", "dany", "cersei", "jamie"] }
+      let(:dv3) { Daru::Vector.new [3,7,0,1,5], index: idx }
+      let(:dv4) { Daru::Vector.new [3,7,0,1,5] }
+
       context "#mean" do
         it "calculates mean" do
           expect(@dv.mean).to eq(282.2)
         end
       end
 
+
+      context "#max" do
+        it "gives the max of a vector" do
+          expect(dv2.max).to eq("tyrion lannister")
+          expect(dv2.max {|i| i.size}).to eq("tyrion lannister")
```

Comment:
1) `dv2.max(&:size)`. 

2) Typically, good specs contain one example per `it`. E.g.
```ruby
context '#max' do
  it 'returns max value' do
    expect(dv2.max).to eq("tyrion lannister")
  end

  it 'allows to provide block' do 
    # ...
  end
end
```

3) It would be better if the second example (with block) return something else than the first. (Now the second example will be "green" even if provided block just ignored inside `max`)